tests/phpunit/tests/admin/includesMediaListTable.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @group admin
+ */
+class Tests_Admin_includesMediaListTable extends WP_UnitTestCase {
+
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+		require_once ABSPATH . 'wp-admin/includes/class-wp-media-list-table.php';
+	}
+
+	/**
+	 * Tests that a call to WP_Media_List_Table::prepare_items() on a site without any scheduled events,
+	 * does not result in a PHP warning.
+	 *
+	 * Warning which we should not see:
+	 * PHP 5.6 - 7.4: `Invalid argument supplied for foreach()`.
+	 * PHP 8.0 and higher: `Warning: foreach() argument must be of type array|object, bool given`.
+	 *
+	 * Note: this doesn't test the actual functioning of the WP_Media_List_Table::prepare_items() method.
+	 * It just and only tests for/against the PHP warning.
+	 *
+	 * @ticket 53949
+	 *
+	 * @covers WP_Media_List_Table::prepare_items
+	 */
+	public function test_prepare_items_without_cron_option_does_not_throw_warning() {
+		global $wp_query;
+
+		// Note: setMethods() is deprecated in PHPUnit 9, but still supported.
+		$mock = $this->getMockBuilder( WP_Media_List_Table::class )
+			->disableOriginalConstructor()
+			->disallowMockingUnknownTypes()
+			->setMethods( array( 'set_pagination_args' ) )
+			->getMock();
+
+		$mock->expects( $this->once() )
+			->method( 'set_pagination_args' );
+
+		$wp_query->query_vars['posts_per_page'] = 10;
```

Comment:
@jrfnl Is setting up `query_vars['posts_per_page']` necessary given that `set_pagination_args` is mocked?

Reply:
@hellofromtonya Yes, it is 100% necessary as the value is passed to the mocked function, but therefore does need to exist. I originally didn't have this and actually tried various options to get round this, but adding this is the only way to get the tests to actually run properly.

Reply:
(well, maybe not the only way, but the most straight forward way I could come up with)

Reply:
And to anticipate your next question - I went with mocking to prevent side-effects from other functions in the function flow of this class from interfering with this encapsulated test for this one particular method.
The method under test does not return anything, but passes off to the next function. By mocking that next function and basically catching that function call instead of letting it run (and setting an expectation for the function call to happen), we can test that the method under tests reaches the end of its processing without erroring out.